components/raftstore/src/store/peer.rs

```diff
@@ -1635,11 +1750,32 @@ impl Peer {
                 return false;
             }
             Ok(RequestPolicy::ReadIndex) => return self.read_index(ctx, req, err_resp, cb),
-            Ok(RequestPolicy::ProposeNormal) => self.propose_normal(ctx, req),
+            Ok(RequestPolicy::ProposeNormal) => {
+                if req.has_admin_request() {
+                    self.propose_batch_request(ctx);
+                    self.propose_normal(ctx, req)
+                } else {
+                    let req_size = req.compute_size();
+                    // No batch request whose size exceed 20% of raft_entry_max_size,
+                    // so total size of request in batch_raft_request would not exceed
+                    // (40% + 20%) of raft_entry_max_size
+                    let max_size = ctx.cfg.raft_entry_max_size.0 as f64 * 0.2;
+                    if self.should_propose_batch_request(ctx, &req) {
+                        self.propose_batch_request(ctx);
+                    }
+                    if f64::from(req_size) < max_size {
+                        self.batch_cmd.push(req, req_size, cb);
```

Comment:
Does this request has the same context with previous reqeusts?

Reply:
I have checked it in `should_propose_batch_request`.